awx/main/models/credential/__init__.py

```diff
@@ -1160,6 +1159,37 @@ def create(self):
     }
 )
 
+ManagedCredentialType(
+    namespace='registry',
+    kind='registry',
```

Comment:
Because the credential is applied to the `podman` process, not the `ansible-playbook` process, I think it does make sense to add a new `kind` option of "registry" for pulling from an image registry.

A big "but" here - this doesn't actually add it as an option. You would need to add it to the `KIND_CHOICES` listing, and this PR does not do that now.

With the way that `ansible-runner` works... this might be accomplish-able by modifying `os.environ`, and not making other ansible-runner code changes. There are still a lot of question marks there.